models/classes/class.QtiTestConverter.php

```diff
@@ -286,7 +286,12 @@ private function arrayToComponent(array $testArray, QtiComponent $parent = null,
                             $this->arrayToComponent($value, $component, true);
                         } else {
                             $assignableValue = $this->componentValue($value, $class);
-                            if (! is_null($assignableValue)) {
+
+                            if ($assignableValue !== null) {
+                                if (is_string($assignableValue) && $key === 'content') {
```

Comment:
You checking whether it is string, but at the line 329 phpDoc says that componentValue method returns `QtiDuration` object or actually it can return `QtiComponentCollection` which is not indicated.

Reply:
```
        if (! is_null($class)) {
            if (is_array($value)) {
                return $this->createComponentCollection(new ReflectionClass($class->name), $value);
            } elseif ($class->name === 'qtism\common\datatypes\QtiDuration') {
                return new qtism\common\datatypes\QtiDuration('PT' . $value . 'S');
            }
        }
        return $value;
```

if `$class` is null it will return provided value without any changes

Reply:
Perhaps it's better to make the code clearly do what you explained:

> if $class is null it will return provided value without any changes

```
if (is_null($class)) {
    return $value;
}
```
Additionally, you can reformat your code to suppress conditional levels and turn to:
```
if (is_array($value)) {
    return $this->createComponentCollection(new ReflectionClass($class->name), $value);
}
if ($class->name === 'qtism\common\datatypes\QtiDuration') {
    return new qtism\common\datatypes\QtiDuration('PT' . $value . 'S');
}
return $value;
```
The duplicate statement with `return $value;` is not an issue, since it offers more readability and ease to test.

Additional point: use imported class for qtism\common\datatypes\QtiDuration instead of strings:
```
use qtism\common\datatypes\QtiDuration;
if ($class->name === QtiDuration::class) {
    return new QtiDuration('PT' . $value . 'S');
}
```